test/widgets/action_sheet_test.dart

```diff
@@ -1061,6 +1061,82 @@ void main() {
       });
     });
 
+    group('MarkTopicAsReadButton', () {
+      Future<void> setupToTopicActionSheetWithUnreadMessages(WidgetTester tester, {
+        ZulipStream? channel,
+      }) async {
+        addTearDown(testBinding.reset);
+
+        final effectiveChannel = channel ?? eg.stream();
+        const topicName = TopicName('test topic');
+        final message = eg.streamMessage(stream: effectiveChannel, topic: 'test topic');
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+          realmUsers: [eg.selfUser],
+          streams: [effectiveChannel],
+          subscriptions: [eg.subscription(effectiveChannel)]));
+        store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        connection = store.connection as FakeApiConnection;
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
```

Comment:
Looking at `setupToMessageActionSheet`, the `connection.prepare` call is right before `tester.pumpWidget`. This is because this call prepares for a request that will be made by `MessageListPage`. Moving it there help highlight their relationship.